modules/nf-core/ribowaltz/templates/ribowaltz.r

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/env Rscript
+
+# Calculation of optimal P-site offsets, diagnostic analysis and visual inspection of ribosome profiling data
+# Author: Ira Iosub
+# Example usage: Rscript ribowaltz.r --bam SRX11780887.Aligned.toTranscriptome.out.bam --gtf Homo_sapiens.GRCh38.111_chr20.gtf --fasta Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz --length_range "26:34" --periodicity_threshold 50 --exclude_start 42 --exclude_stop 27
+
+suppressPackageStartupMessages(library(riboWaltz))
+suppressPackageStartupMessages(library(dplyr))
+
+# Allow data.table to use all available threads
+data.table::setDTthreads(0)
+
+# =========
+# Options and paths
+# =========
+
+
+#' Check for Non-Empty, Non-Whitespace String
+#'
+#' This function checks if the input is non-NULL and contains more than just whitespace.
+#' It returns TRUE if the input is a non-empty, non-whitespace string, and FALSE otherwise.
+#'
+#' @param input A variable to check.
+#' @return A logical value: TRUE if the input is a valid, non-empty, non-whitespace string; FALSE otherwise.
+#' @examples
+#' is_valid_string("Hello World") # Returns TRUE
+#' is_valid_string("   ")         # Returns FALSE
+#' is_valid_string(NULL)          # Returns FALSE
+
+is_valid_string <- function(input) {
+    !is.null(input) && nzchar(trimws(input))
+}
+
+#' Parse out options from a string without recourse to optparse
+#'
+#' @param x Long-form argument list like --opt1 val1 --opt2 val2
+#'
+#' @return named list of options and values similar to optparse
+
+parse_args <- function(x){
+    args_list <- unlist(strsplit(x, ' ?--')[[1]])[-1]
+    args_vals <- lapply(args_list, function(x) scan(text=x, what='character', quiet = TRUE))
+
+    # Ensure the option vectors are length 2 (key/ value) to catch empty ones
+    args_vals <- lapply(args_vals, function(z){ length(z) <- 2; z})
+
+    parsed_args <- structure(lapply(args_vals, function(x) x[2]), names = lapply(args_vals, function(x) x[1]))
+    parsed_args[! is.na(parsed_args)]
+}
+
+
+# Function to print help menu
+print_help_menu <- function() {
```

Comment:
This is never going to be run interactively, I don't think the inline help makes sense to include.

Reply:
thanks! done. Referencing commit 2c16f8e.